yourbench/pipeline/summarization.py

```diff
@@ -0,0 +1,497 @@
+            for the same prediction).
+
+    Returns:
+        float: The overall BLEU score for the corpus. Returns 0.0 if an error
+        occurs or if inputs are empty.
+    """
+    try:
+        safe_preds = [p if p else "" for p in predictions]
+        safe_refs = []
+        for ref_list in references:
+            if not isinstance(ref_list, list) or not ref_list:
+                safe_refs.append([""])
+            else:
+                safe_refs.append([r if r else "" for r in ref_list])
+
+        if not safe_preds or not safe_refs:
+            logger.warning("Skipping BLEU due to empty inputs.")
+            return 0.0
+
+        bleu_metric = evaluate.load("bleu")
+        bleu_result = bleu_metric.compute(predictions=safe_preds, references=safe_refs)
+        return bleu_result.get("bleu", 0.0)
+    except Exception as e:
+        logger.error("Error computing BLEU score: {}", str(e))
+        return 0.0
+
+
+def _safe_compute_meteor_score(predictions: List[str], references: List[str]) -> float:
+    """
+    Compute METEOR score safely with error handling.
+
+    Args:
+        predictions (List[str]): The candidate summaries or predicted texts.
+        references (List[str]): The reference texts matching each prediction.
+
+    Returns:
+        float: The corpus-level METEOR score (0.0 to 1.0). Returns 0.0 if an
+        error occurs or if inputs are empty.
+    """
+    try:
+        safe_preds = [p if p else "" for p in predictions]
+        safe_refs = [r if r else "" for r in references]
+
+        if not safe_preds or not safe_refs:
+            logger.warning("Skipping METEOR due to empty inputs.")
+            return 0.0
+
+        meteor_metric = evaluate.load("meteor")
+        meteor_result = meteor_metric.compute(predictions=safe_preds, references=safe_refs)
+        return meteor_result.get("meteor", 0.0)
+    except Exception as e:
+        logger.error("Error computing METEOR score: {}", str(e))
+        return 0.0
+
+
+def _safe_compute_rouge_scores(predictions: List[str], references: List[str]) -> Dict[str, float]:
+    """
+    Compute ROUGE scores safely with error handling.
+
+    Args:
+        predictions (List[str]): Candidate summaries or predicted texts.
+        references (List[str]): Reference texts matching each prediction.
+
+    Returns:
+        Dict[str, float]: Contains 'rouge1', 'rouge2', and 'rougeL' scores.
+        Returns a dictionary of zeros if an error occurs or if inputs are empty.
+    """
+    try:
+        safe_preds = [p if p else "" for p in predictions]
+        safe_refs = [r if r else "" for r in references]
+
+        if not safe_preds or not safe_refs:
+            logger.warning("Skipping ROUGE due to empty inputs.")
+            return {"rouge1": 0.0, "rouge2": 0.0, "rougeL": 0.0}
+
+        rouge_metric = evaluate.load("rouge")
+        rouge_result = rouge_metric.compute(predictions=safe_preds, references=safe_refs)
+        return {
```

Comment:
in which case can you get 0 in rouge results? Shouldn't you print a warning?

Reply:
removed!